src/timeout.ts

```diff
@@ -323,6 +323,15 @@ export class CSOTTimeoutContext extends TimeoutContext {
     return remainingTimeMS;
   }
 
+  clone(): CSOTTimeoutContext {
```

Comment:
Can we also add a doc comment here explaining that this is intended to be used in situations where we need to perform concurrent operations but operating on the same deadline, but cannot share a single TimeoutContext.